rebasehelper/specfile.py

```diff
@@ -717,31 +723,32 @@ def set_extra_version(self, extra_version):
 
                 preamble = self.spec_content.section('%package')
 
-                # change the Source0 definition
-                source0_re = re.compile(r'^Source0?\s*:.+')
+                # change the main source definition
+                main_source_re = re.compile(r'^Source0?\s*:.*$' if self.main_source_index == 0 else
+                                            r'^Source{0}\s*:.*$'.format(self.main_source_index))
                 for index, line in enumerate(preamble):
-                    if source0_re.search(line):
-                        # comment out the original Source0 line
-                        logger.verbose("Commenting out original Source0 line '%s'", line.strip())
+                    if main_source_re.match(line):
+                        # comment out the original main source line
+                        logger.verbose("Commenting out original main source line '%s'", line.strip())
```

Comment:
Perhaps it could be good to mention the number of the source in the logged message?

Reply:
The number of the source is included in `line`.